addons/hr_attendance/wizard/hr_attendance_error.py

```diff
@@ -2,47 +2,33 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 import time
 
-from openerp.osv import fields, osv
-from openerp.tools.translate import _
+from openerp import api, fields, models, _
 from openerp.exceptions import UserError
 
 
-class hr_attendance_error(osv.osv_memory):
+class HrAttendanceError(models.TransientModel):
 
     _name = 'hr.attendance.error'
     _description = 'Print Error Attendance Report'
-    _columns = {
-        'init_date': fields.date('Starting Date', required=True),
-        'end_date': fields.date('Ending Date', required=True),
-        'max_delay': fields.integer('Maximum Tolerance (in minutes)', required=True,
-            help="Allowed difference in minutes between the signin/signout and the timesheet computation for one sheet. Set this to 0 for no tolerance.")
-    }
-    _defaults = {
-         'init_date': lambda *a: time.strftime('%Y-%m-%d'),
-         'end_date': lambda *a: time.strftime('%Y-%m-%d'),
-         'max_delay': 120,
-    }
 
-    def print_report(self, cr, uid, ids, context=None):
+    init_date = fields.Date('Starting Date', required=True, default=lambda *a: time.strftime('%Y-%m-%d'))
+    end_date = fields.Date('Ending Date', required=True, default=lambda *a: time.strftime('%Y-%m-%d'))
+    max_delay = fields.Integer('Maximum Tolerance (in minutes)', required=True, default=120,
+                help="Allowed difference in minutes between the signin/signout and the timesheet computation for one sheet. Set this to 0 for no tolerance.")
+
+    @api.multi
+    def print_report(self):
         emp_ids = []
-        data_error = self.read(cr, uid, ids, context=context)[0]
-        date_from = data_error['init_date']
-        date_to = data_error['end_date']
-        cr.execute("SELECT id FROM hr_attendance WHERE employee_id IN %s AND to_char(name,'YYYY-mm-dd')<=%s AND to_char(name,'YYYY-mm-dd')>=%s AND action IN %s ORDER BY name" ,(tuple(context['active_ids']), date_to, date_from, tuple(['sign_in','sign_out'])))
-        attendance_ids = [x[0] for x in cr.fetchall()]
+        date_from = self.init_date
+        date_to = self.end_date
+        self.env.cr.execute("SELECT id FROM hr_attendance WHERE employee_id IN %s AND to_char(name, 'YYYY-mm-dd')<=%s AND to_char(name, 'YYYY-mm-dd')>=%s AND action IN %s ORDER BY name", (tuple(self.env.context['active_ids']), date_to, date_from, tuple(['sign_in', 'sign_out'])))
+        attendance_ids = [x[0] for x in self.env.cr.fetchall()]
         if not attendance_ids:
             raise UserError(_('No records are found for your selection!'))
-        attendance_records = self.pool.get('hr.attendance').browse(cr, uid, attendance_ids, context=context)
-
+        attendance_records = self.env['hr.attendance'].browse(attendance_ids)
         for rec in attendance_records:
             if rec.employee_id.id not in emp_ids:
```

Comment:
NO, due to SET no doublication of record, so no need of this 
if rec.employee_id.id not in emp_ids: condition